api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -177,4 +191,26 @@ default Long fileSequenceNumber() {
   default F copy(boolean withStats) {
     return withStats ? copy() : copyWithoutStats();
   }
+
+  /**
+   * Copies this file (potentially with or without specific column stats). Manifest readers can
+   * reuse file instances; use this method to copy data when collecting files from tasks.
+   *
+   * @param withStats Will copy this file without file stats if set to <code>false</code>.
+   * @param statsToKeep Will keep stats only for these columns. Not used if <code>withStats</code>
+   *     is set to <code>false</code>.
+   * @return a copy of this data file. If "withStats" is set to <code>false</code> the file will not
+   *     contain lower bounds, upper bounds, value counts, null value counts, or nan value counts.
+   *     If "withStats" is set to <code>true</code> and the "statsToKeep" is not empty then only
+   *     specific column stats will be kept.
+   */
+  default F copy(boolean withStats, Collection<Integer> statsToKeep) {
```

Comment:
is this needed in this interface? `copyWithSpecificStats` from line 177 should imply `withStats` to be true.

Reply:
I have found nicer to have a default method, than doing the calculation in `BaseDistributedDataScan.toFileTasks` and in `ManifestGroup.createFileScanTasks` too.

I am ok with moving this to the methods mentioned above if you think it is more important to keep the interface thin. Please leave  comment, if you still think we should move the logic.

Reply:
yeah. I think it is more important to keep the interface lean. I am wondering if we can move them into a util class like `ContentFileUtil` that is similar to `ManifestFileUtil`.

Reply:
Done